### PR TITLE
Document gofmt automatic formatting guidelines

### PR DESCRIPTION
**Pull request by ignasi35**
I was getting tired of looking for the correct command each time.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,9 @@
 lint:
 	golangci-lint run
 
+format:
+	gofmt -w .
```
Comment (ignasi35) [inline Makefile:5]: I've seen some places (including GitHub Copilot) suggest `gofmt -s -w .` (with `-s`).

That supposedly simplifies code (whatever `simplify` means). I'd like to hear the opinion of someone with more Go experience than me on the convenience of using `-s`.

Comment (gonzaloserrano) [reply Makefile:5]: I think simplify does thinks like converting `var i int = 1` to `i := 1`.

We can add it. I even prefer gofumpt, that does eve more things... but that's another topic.

**Comment by gonzaloserrano** [comment]
> I was getting tired of looking for the correct command each time.

Your editor/IDE should run `gofmt -w -s` on file save.

**Comment by ignasi35** [comment]
Good point.

In that case I'll document it on CONTRIBUTING (if not done already).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+
+## IDE settings
+
+1. Make sure your IDE runs `gofmt -w -s` on file save.
+
+2. Make sure your IDE adds a blank line at the end of the file.
+
+### VSCode
+
+1. To format on save, isntall the [Go extension for VSCode](https://code.visualstudio.com/docs/languages/go) and go to settings:
```
Comment (gonzaloserrano) [inline CONTRIBUTING.md:20]:
```suggestion
1. To format on save, install the [Go extension for VSCode](https://code.visualstudio.com/docs/languages/go) and go to settings:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+    1. in `Go: Format Flags` add `-w -s`
+    1. in `Go: Format Tools` select `gofmt`
+
+2. To insert a newlineon save, go to settings:
```
Comment (gonzaloserrano) [inline CONTRIBUTING.md:24]:
```suggestion
2. To insert a new line at the end of file on save, go to settings:
```

**Review by gonzaloserrano — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (ignasi35) [reply CONTRIBUTING.md:20]: I can't believe the amount of times I make that typo. My memory muscle or something is deeply broken.

File: CONTRIBUTING.md (hunk 3 above)
Comment (gonzaloserrano) [reply CONTRIBUTING.md:24]: I think (not sure since I'm on mobile) that this suggestion was not applied.

**Comment by gonzaloserrano** [comment]
PR title should be changed so when merging it, as it's a multiple commit one, the squashed commit will have the title.

**Comment by ignasi35** [comment]
> PR title should be changed so when merging it, as it's a multiple commit one, the squashed commit will have the title.

Good catch.

Done

File: CONTRIBUTING.md (hunk 3 above)
Comment (ignasi35) [reply CONTRIBUTING.md:24]: I think I messed up laptops (I did the changes on the other one and didn't push them). 🤦🏼‍♂️

**Pull request merged**